tests/test_auto_attribs__py3.py

```diff
@@ -0,0 +1,54 @@
+from typing import Optional
+
+import attr
+import pytest
+
+from attrs_strict import type_validator
+
+
+@pytest.mark.parametrize(
+    "typ, good_value, bad_value",
+    [(str, "str", 0xBAD), (int, 7, "bad"), (Optional[str], None, 0xBAD)],
+)
+def test_real_types(typ, good_value, bad_value):
```

Comment:
Is the argument here `typ` intended or a typo ? :-) I'd prefer `type_` if you want to avoid name clashes :-)

Reply:
Trying to avoid collisions.  I've also seen `tp` being used (in typing.py) but I'll use `type_` since I see you're using that already.